packages/crashlytics/ios/Plugin/FirebaseCrashlyticsPlugin.swift

```diff
@@ -108,20 +111,39 @@ public class FirebaseCrashlyticsPlugin: CAPPlugin, CAPBridgedPlugin {
     }
 
     @objc func recordException(_ call: CAPPluginCall) {
-        guard let message = call.getString("message") else {
-            call.reject(errorMessageMissing)
-            return
+      guard let message = call.getString("message") else {
+        call.reject(errorMessageMissing)
+        return
+      }
+      
+      let stacktraceOptional = call.getArray("stacktrace")
+      let customPropertiesOptional = call.getArray("customProperties")
+        
+      guard !(stacktraceOptional != nil && customPropertiesOptional != nil) else {
+        call.reject(errorStacktraceAndCustomPropertiesDisallowed)
+        return
+      }
+      
+      if(stacktraceOptional != nil) {
+        guard let stacktrace = stacktraceOptional as? [JSObject] else {
+          call.reject(errorStacktraceCastFailed)
+          return
         }
-
-        let stacktrace = call.getArray("stacktrace", JSObject.self)
-        if stacktrace == nil || stacktrace!.isEmpty {
-            let domain = call.getString("domain") ?? ""
-            let code = call.getInt("code") ?? -1001
-
-            implementation?.recordException(message, domain, code)
+        implementation?.recordExceptionWithStacktrace(message, stacktrace)
+      } else {
+        let domain = call.getString("domain") ?? ""
+        let code = call.getInt("code") ?? -1001
+        
+        if(customPropertiesOptional != nil) {
+          guard customPropertiesOptional != nil ,let customProperties = customPropertiesOptional as? [JSObject] else {
+            call.reject(errorCustomPropertiesCastFailed)
+            return
+          }
```

Comment:
You can also remove this check. That's what the TS types are for. I want to keep the plugin as simple as possible. If you force an invalid type, the value should be ignored.